core/src/main/java/com/netflix/iceberg/util/BinPacking.java

```diff
@@ -119,6 +127,18 @@ public boolean hasNext() {
       return ImmutableList.copyOf(bins.removeFirst().items());
     }
 
+    private Bin getLargestBin() {
```

Comment:
The name should be `removeLargestBin` because it modifies `bins`. I think this should also be a `static` method with `bins` passed in as an argument. That makes it clear what variable is side-effected by the method because it can only change variables that are passed in.

Reply:
@rdblue - I've updated the method name and argument list. However, within the current implementation, making the method `static` requires making `Bin` and `targetWeight` static as well. I did not think that was the right direction of this PR but please advise if you still want me to go down that route.

Reply:
Sounds fine to me.